src/commands/positive.ts

```diff
@@ -250,20 +256,27 @@ export default class Positive extends Command {
           scheme: scheme.scheme,
         };
       });
-    // Wrap if statements in loop to iterate over securities array to check for apiKey or bearer_token(s)
+
     let apiKey = flags.apiKey;
     let token = flags.bearerToken;
+    const noPrompt = flags.noPrompt;
     for (const security of securities) {
       if (security.type === OASSecurityType.APIKEY) {
         const apiSecurityName = security.key;
-        const apiKeyValue =
-          apiKey ??
-          // eslint-disable-next-line no-await-in-loop
-          (await cli.prompt('Please provide your API Key', {
-            type: 'mask',
-          }));
+        let apiKeyValue;
+        if (!noPrompt) {
+          apiKeyValue =
+            apiKey ??
+            // eslint-disable-next-line no-await-in-loop
+            (await cli.prompt('Please provide your API Key', {
+              type: 'mask',
+            }));
+        }
+        // if apiKey is undefined, throw an error and provide a message to the user
+        apiKeyValue = apiKey;
         if (!apiKey) {
           apiKey = apiKeyValue;
+          this.log('API key is undefined or empty.');
         }
 
         this.securityValues[apiSecurityName] = { value: apiKeyValue };
```

Comment:
```suggestion
        if (!noPrompt && !apiKey) {
          apiKey = // eslint-disable-next-line no-await-in-loop
            (await cli.prompt('Please provide your API Key', {
              type: 'mask',
            }));
          }
 
        // if apiKey is undefined, throw an error and provide a message to the user
        if (!apiKey) {
          this.log('API key is undefined or empty.');
        }

        this.securityValues[apiSecurityName] = { value: apiKey };
```

Reply:
The formatting of this suggestion is bad, but I think we need something like this.

The way it's currently written if the API key flag isn't passed in, then the API key value stays undefined even if the user provides one